.github/workflows/build-and-publish.yml

```diff
@@ -36,21 +36,19 @@ jobs:
         with:
           path: ~/.m2
           key: ${{ runner.os }}-m2-${{ hashFiles('**/pom.xml') }}
-          restore-keys: ${{ runner.os }}-m2
+          restore-keys: ${{ runner.os }}-m2-
       - name: Set package version
         id: set-package-version
         run: |
           PACKAGE_VERSION="1.0.$(date +"%Y%m%d%H%M%S")"
           echo "PACKAGE_VERSION=$PACKAGE_VERSION" >> "$GITHUB_OUTPUT"
-          mvn versions:set -DnewVersion="$PACKAGE_VERSION"
-
+          mvn -B --no-transfer-progress versions:set -DnewVersion="$PACKAGE_VERSION"
       - name: Release Maven package
-        run: mvn deploy --batch-mode --settings "/home/runner/.m2/settings.xml"
```

Comment:
Fine to try without the settings path - it shouldn't be needed in theory (I can't remember if we previously got to the state where it worked for us without it)